### ISSUE TITLE
False-positive `RedundantFunctionCall` on `ksort()` on a merged array

### ISSUE THREAD

**Issue by lukasbestle** [vimeo/psalm#8186]
`array_merge()` keeps the key order of the merged arrays. Yet, Psalm flags the use of `ksort()` as redundant. But if I were to remove `ksort()`, the `$merged` array will have a different key order.

```php
$a = ['c' => 'C', 'a' => 'A'];
$b = ['b' => 'B', 'd' => 'D'];

$merged = array_merge($a, $b);

ksort($merged);
```

https://psalm.dev/r/e4ee2f0520

**Comment by weirdan** [comment]
No longer reproducible
